### PR TITLE
User-state fixes et al

### PR DESCRIPTION
**Pull request by sjmiles**
Still working on fixes for mistakes outlined in #1006 (in particular see the chart about Accidental Sources of Truth).

See also #1002.

### CODE REVIEW

File: shell/artifacts/Common/source/MasterDetail.js [hunk 1]
```diff
@@ -51,7 +52,7 @@ defineParticle(({DomParticle}) => {
 </style>
 <div ${host}>
   <div class="detail-wrapper" style%="{{tab0}}">
-    <button on-click="_onBack" class="close-button material-icons">close</button>
```
Comment (shaper) [inline shell/artifacts/Common/source/MasterDetail.js]: Just making sure, you intended to have this and the other visual/style changes below as part of this PR?

File: shell/artifacts/Restaurants/source/RestaurantDetail.js [hunk 2]
```diff
@@ -6,11 +6,11 @@
 // subject to an additional IP rights grant found at
 // http://polymer.github.io/PATENTS.txt
 
-defineParticle(({DomParticle}) => {
+defineParticle(({DomParticle, html}) => {
 
   let host = `restaurant-detail`;
 
-  let styles = `
+  let styles = html`
```
Comment (shaper) [inline shell/artifacts/Restaurants/source/RestaurantDetail.js]: This file -- intentional for this PR?

File: shell/artifacts/Common/source/MasterDetail.js (hunk 1 above)
Comment (sjmiles) [reply shell/artifacts/Common/source/MasterDetail.js]: I changed icon handling in one of the other recent PR's, so I felt like I should fix it right away. But I didn't actually mean to slip the fixes into this PR. I think we might as well carry on at this point.

File: shell/artifacts/Restaurants/source/RestaurantDetail.js (hunk 2 above)
Comment (sjmiles) [reply shell/artifacts/Restaurants/source/RestaurantDetail.js]: There was style leakage happening from MasterDetail to RestaurantDetail. When I fixed the former, I had to fix the latter. Neither should be in this exact PR, but they are still good changes.

**Pull request merged**